### ISSUE TITLE
About training time cost

### ISSUE THREAD

**Issue by joshua19881228** [nashory/progressive-growing-torch#13]
It's a great implementation. And I can see that it only costed 12 minute to get 91872/202599.  I wonder what your hardware setting is because it took about 1 hour to get 1568/596385 in my own 4 Titan xps. Or is there anything wrong with my experiment settings?

**Comment by nashory** [comment]
Yes, it should be extremely fast because multi-threaded dataloading is supported in my code.
In addition, in the very early stage, the resolution is only 4x4, tiny images.
However, I guess when the training goes on, and you starts to load high resolution images (e.g. 512x512 or 1024x1024),  the training time could slow down. (The speed for 1 iteration could be same, as the batchsize for 1 iteration becomes small.)

I had really hard time to reproduce the result, and now the code seems to be working properly.
my current exp. setting is:
learningrate = 0.0001
transition_tick = 300
training_tick = 300
epsilon_drift = 0.001
adam optimizer {
   - beta1 = 0.0,
   - beta2 = 0.99,
   - epsilon_drift = 0.001,
}

I will post the very promising result soon.
(The transition from 4x4 to 8x8 is quiet successful, and waiting for the further result now.)

Thank you

**Comment by nashory** [comment]
You can check my hardward setting described in README though :)

**Comment by joshua19881228** [comment]
@nashory Thanks. It ran much faster after I re-run the run.py. The original images are collected by parsing from the internet and the faces are detected and rotated to 140*140. I planned to use progressive-growing-gan to generate other kind of images. But for now I wanted to verify whether it can generate faces because in my understanding, faces are easier to generate.

It is really a great work of you implementing pggan. I'll also share my progress to you if they are helpful for your implementing.

**Comment by nashory** [comment]
@joshua19881228 
yes, it will be greatly helpful more contributor like you tries to fix bugs by running in different environment. Since training this PGGAN is very time consuming which makes harder to debug.
good luck! :)

**Comment by joshua19881228** [comment]
I've got a 16*16 result. ![16860_gtrns_100.0_0.0](https://github.com/joshua19881228/joshua19881228.github.io/blob/master/_drafts/16860_gtrns_100.0_0.0.jpg) It seems to be a failure.... One of my sample is like this ![26_male_1668.jpg_0.png](https://github.com/joshua19881228/joshua19881228.github.io/blob/master/_drafts/26_male_1668.jpg_0.png)

**Comment by nashory** [comment]
yeah, It's pretty much hard to transfer from 8x8 to 16x16.
I found LRN(1) un my code is actually harmful (not sure but maybe.)
I just removed all LRN and tanh layers. I figure PGGAN is extremely sensitive to network structure, and since the authors made slight changes to the existing layers(e.g. pixelwise norm / minibatch discrimination, generalied dropout etc.), It is hard to reconstruct exactly same structure the paper used.
I will post the result if I success to generate 16x16 image.

**Comment by joshua19881228** [comment]
@nashory In the paper, there is a sentence "we represent training and generated images in [-1,1]". I am not sure whether it means that we should re-arrange the pixel value to [-1,1] or you have already done that in the code.

**Comment by nashory** [comment]
dataloader converts the range into [-1,1,].
That's why I could use Tanh() for the generator :)
you can check preprocess.adjust_range function in preprocess.lua file!


2017. 11. 9. 오후 8:42에 "Joshua Li" <notifications@github.com>님이 작성:
